apps/system/test/unit/lockscreen/widgets/clock/lockscreen_clock_widget_test.js

```diff
@@ -49,4 +49,50 @@ suite('LockScreenClockWidget > ', function() {
       mockThis.resources.elements.date.textContent,
       `it doesn't update the date.textContent with the date in locale format`);
   });
+
+  test(`it would update the alarm info (an alarm is set)`, function() {
```

Comment:
Without cooperating `done` with `catch` and `assert`, the test will always pass, since after the promise kicks off, the stack is empty and the test is end: it never wait the test result in the async chain.